index/segment/mem/types.go

```diff
@@ -62,3 +54,75 @@ type Options interface {
 	// NumInsertWorkers returns the number of insert workers.
 	NumInsertWorkers() int
 }
+
+// immutableDocIDsSet is a memory efficient collection of docIDs. It only permits
+// for read-only operations.
+type immutableDocIDsSet interface {
```

Comment:
Seems like this is what is traditionally called a postings list, whereas the `postingsList` interface below is a manager of different postings lists. Is it worth updating the terminology perhaps?

Reply:
Sure thing